frontend/app/activeGame.js

```diff
@@ -1,11 +1,138 @@
-export default function activeGame(gameID) {
-    // my dummy active game to display board.
-    /* eslint no-console: ["error", { allow: ["log"] }] */
-    console.log(`Active game called with ${gameID}`);
-    document.getElementById('content').innerHTML = '<canvas id="myCanvas" width="450" height="450" style="border:1px solid #000000;">';
-    const c = document.getElementById('myCanvas');
+import * as generateUserDetails from './generateUserDetails';
+import * as control from './moveFunctions';
+import * as getCookie from './checkUserIDCookie';
+import {getEventSource} from './sse';
+import * as sendJSON from './sendJSON';
+
+let details = getCookie.checkUserDetails();
+let id = details.user_id;
+
+/**
+ * Displays the page for an active game.
+ *
+ * @param gameID The ID for the game that will be displayed.
+ */
+export function activeGame(gameID, playerList) {
+    // display board and assign starting positions.
+    displayBoard(playerList);
+    generateUserDetails.generateUserDetails();
+    details = getCookie.checkUserDetails();
+    id = details.user_id;
+    enableActiveGameListeners();
+}
+
+
+/**
+ * Called when a playerMove event happens.
+ * Moves the player location on the board using the received data.
+ *
+ * @param playerMoveEvent The data received from the event
+ */
+export function onPlayerMove(playerMoveEvent) {
+    const move = String(JSON.parse(playerMoveEvent.data));
```

Comment:
I thought the playerMoveEvent data was an array of moves, where the first element was the user id and the second was the new position? If so, I don’t really follow the call to `String()`.

Reply:
This I don't remember why honestly. I'll look into it.

Reply:
Without the conversion into String the if statement below was only working if I used ==, which we can't use with our testing. Hence the conversion. I've moved it into the if statement itself now.

Reply:
Ok, that sounds fine. We are having problems with types though, we should figure this out at some point.